src/vs/code/electron-sandbox/issue/issueReporterService.ts

```diff
@@ -1168,20 +1196,45 @@ export class IssueReporter extends Disposable {
 			this.addEventListener('extension-selector', 'change', async (e: Event) => {
 				this.clearExtensionData();
 				const selectedExtensionId = (<HTMLInputElement>e.target).value;
+				this.changeExtension = selectedExtensionId;
 				const extensions = this.issueReporterModel.getData().allExtensions;
 				const matches = extensions.filter(extension => extension.id === selectedExtensionId);
 				if (matches.length) {
 					this.issueReporterModel.update({ selectedExtension: matches[0] });
 					const selectedExtension = this.issueReporterModel.getData().selectedExtension;
-					if (selectedExtension) {
-						selectedExtension.data = undefined;
-						selectedExtension.uri = undefined;
+					if (selectedExtension && !this.loadingExtensionData) {
+						const iconElement = document.createElement('span');
+						iconElement.classList.add(...ThemeIcon.asClassNameArray(Codicon.loading), 'codicon-modifier-spin');
+						this.setLoading(iconElement);
+						const openReporterData = await this.sendReporterMenu(selectedExtension);
+						if (openReporterData && (this.changeExtension === selectedExtensionId)) {
+							this.removeLoading(iconElement, true);
+							this.configuration.data = openReporterData;
+						} else if (openReporterData && this.changeExtension !== selectedExtensionId) {
+							this.removeLoading(iconElement, this.openReporter);
```

Comment:
you pass in `this.openReporter` here but I noticed that `this.removeLoading` takes in that value and then immediately sets:
```ts
this.openReporter = fromReporter
```
and that feels unintentional. Especially without a comment why.

Also, this makes me wonder about this scenario:
* Select extension 1
* Quickly Select extension 2 before Extension 1 comes back
* Extension 1 finishes in 5 seconds
* Extension 2 finishes in 20 seconds

based on the code here, if Im reading it right, I feel like when Extension 1 finishes, it will remove the loading... even though we switched to Extension 2 and the loading should still happen until Extension 2 finishes